code/modules/mob/living/carbon/carbon.dm

```diff
@@ -1318,3 +1318,21 @@
 	to_chat(src, span_danger("You shove [target.name] into [name]!"))
 	log_combat(src, target, "shoved", "into [name]")
 	return COMSIG_CARBON_SHOVE_HANDLED
+
+
+// Checks to see how many hands this person has to sign with.
+/mob/living/carbon/proc/check_signables_state()
+	var/mob/living/carbon/mute = src
+	var/obj/item/bodypart/left_arm = mute.get_bodypart(BODY_ZONE_L_ARM)
+	var/obj/item/bodypart/right_arm = mute.get_bodypart(BODY_ZONE_R_ARM)
+	var/empty_indexes = get_empty_held_indexes()
+	if(length(empty_indexes) == 0 || (length(empty_indexes) < 2 && ((!left_arm || left_arm.bodypart_disabled) || (!right_arm || right_arm.bodypart_disabled))))//All existing hands full, can't sign
+		return 1 // These aren't booleans
```

Comment:
You should make these into defines if they're not booleans
like `define VALID_SIGNER 0`, `define FULL_HANDS 1`, `define NO_ARMS 2`, etc

Reply:
Would I not have to define those for the other files too? Or do you mean for me to add those to the actual _DEFINES/mobs.dm file?

Reply:
Hope that was what you meant, I swapped it

Reply:
LGTM